consensus/src/ordered_broadcast/mod.rs

```diff
@@ -61,26 +52,31 @@ cfg_if::cfg_if! {
 #[cfg(test)]
 pub mod mocks;
 
-/// Used as the [`Index`](crate::Supervisor::Index) type.
-/// Defines the current set of sequencers and validators.
-///
-/// This is not a single "View" in the sense of a consensus protocol, but rather a continuous
-/// sequence of views in-which the set of sequencers and validators is constant.
-pub type Epoch = u64;
-
-/// Used as the [`Automaton::Context`](crate::Automaton::Context) type.
-#[derive(Debug, Clone, Hash, PartialEq, Eq)]
-pub struct Context<P: Array> {
-    /// Sequencer's public key.
-    pub sequencer: P,
-
-    /// Sequencer-specific sequential height. Zero-indexed.
-    pub height: u64,
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Decode error: {0}")]
+    Decode(#[from] prost::DecodeError),
+    #[error("Missing chunk")]
+    MissingChunk,
+    #[error("Missing parent")]
+    ParentMissing,
+    #[error("Parent on genesis chunk")]
+    ParentOnGenesis,
+    #[error("Invalid partial")]
+    InvalidPartial,
+    #[error("Invalid threshold")]
+    InvalidThreshold,
+    #[error("Invalid sequencer")]
+    InvalidSequencer,
+    #[error("Invalid payload")]
+    InvalidPayload,
+    #[error("Invalid signature")]
+    InvalidSignature,
 }
```

Comment:
should these errors be in `types.rs`? Not clear to me that other files currently use them, though we could maybe use them in `engine.rs`?

Reply:
I think the errors should be unified (looks like I need to get rid of the Prost reference there too...)